src/network/core/address.cpp

```diff
@@ -400,17 +399,40 @@ void NetworkAddress::Listen(int socktype, SocketList *sockets)
 	}
 }
 
+/**
+ * Get the peer address of a socket as NetworkAddress.
+ * @param sock The socket to get the peer address of.
+ * @return The NetworkAddress of the peer address.
+ */
+/* static */ NetworkAddress NetworkAddress::GetPeerAddress(SOCKET sock)
+{
+	sockaddr_storage addr = {};
+	socklen_t addr_len = sizeof(addr);
+	getpeername(sock, (sockaddr *)&addr, &addr_len);
```

Comment:
Not sure whether it is likely that this function returns an error, but I think it would be better to at least log the error and return an "empty" network address than one composed of garbage data from the stack.